src/Data/Array.idr

```diff
@@ -99,6 +100,33 @@ take k (A size arr) with (k <= size) proof eq
   _ | True  = A k $ take k arr @{lteOpReflectsLTE _ _ eq}
   _ | False = A size arr
 
+partial export %inline
+drop : Nat -> Array a -> Array a
```

Comment:
I think, this overly complicated. See my comment on `take` for indexed arrays. This should actually just invoke that.

Reply:
Ah okay, sounds good, I think this will be easy once I work through the `drop` implementation you're helping with below in `Data.Array.Core`.